### PR TITLE
Add language statistics and improvements to output readability

### PR DESCRIPTION
**Pull request by alecello**
## Language statistics
Generate language statistics for each repository and globally, to show how much of our code is made by each language, in percentage.

### How
- GitHub's api (https://api.github.com/repos/{owner}/{repo}/languages)
- CLOC's native language statistics

Both report a list (sorted in descending order) of langauges used in a given dataset: GitHub reports the number of bytes whereas CLOC reports number of lines, but since we're interested in percentages, both are fine

The result is the generation of a new pie chart for each repository (plus a global one) with the language distribution.

### When
Language statistics are always generated when SLOCs are counted with `cloc`, since in that case those statistics are basically free by further parsing CLOC's output.

If SLOCSs are counted with `wc` or not counted at all, the user can specify the new `--lang` command line argument to have the script fetch those informations from GitHub's APIs. Of course I also added a complementary `--no-lang` that skips language generation.

Note that since CLOC's output is much more detailed than GitHub's APIs, it always takes precedence: if CLOC is being used then the `--lang` and `--no-lang` options are ignored and if neither is provided, the prompt asking the user wether they want to generate language statistics is skipped.

## Documentation
I updated the combined.svg used in the `README` and removed the other two unused SVGs
Also i changed the `README` to reflect the new command line arguments. I also made some somewhat random changes here and there, the bigger one being in the usage instructions. These changes have various reasons behind them but are mostly subjective and stylistic, so have a look at them and let me know if the changes are appreciated.

## Output
I made some changes to the output the program produces:
- I added an output section for the language statistics
- Hid `git clone` output for cleaner interaction with the user
- Added counters to graph generation so the user can see how it's coming along (it can take a while on slower machines)
- Changed output so that each "section" (contributions, commits, SLOC, graphs, etc.) is separated by two newlines from the previous one and all tasks within a section (counting SLOC for a specific repo etc.) are indented
- Changed report output such that global statistics are separated by per-repo statistics in each section by a newline, to make them stand out more

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -24,19 +24,21 @@ def __init__(self,
...
         self.legend = legend
         self.title = title
+        self.counter = counter
         
         self.data = _normalize_data(data, minimum)
         self.count = len(self.data)
 
     def is_suitable(self) -> bool:
-        return True if len(self.data) >= self.min_count else False
+        return True if self.count >= self.min_count else False
```
Comment (e-caste) [inline main.py]: Unnecessary ternary condition
```suggestion
        return self.count >= self.min_count
```

File: main.py [hunk 2]
```diff
@@ -109,15 +112,16 @@ def get_anonymous_commits_stats(repos: list, header: dict) -> dict:
...
 
             if response.status_code == 403:
                 raise_rate_limited_exception()
             elif response.status_code == 200:
                 stats[repo] = sum([weekly['total'] for weekly in response.json()])
                 stats['total'] += stats[repo]
+
         else:
-            print(f'Using cache for repo {repo}')
+            print("\t" f"{i + 1}/{len(repos)} - {repo} - Using cached result...")
```
Comment (e-caste) [inline main.py]: Use only one string when two are not necessary. See also lines 115, 154, 167, 338, 483
```suggestion
            print(f"\t{i + 1}/{len(repos)} - {repo} - Using cached result...")
```

File: main.py [hunk 3]
```diff
@@ -545,7 +645,22 @@ def print_all_stats(commits_stats: dict, lines_stats: dict, contributors_stats:
...
-    output = "\n\n".join([contributors_output, '*' * 42, commits_output, '*' * 42, lines_output])
+    if language_repo is not None:
+        language_output = 'Language usage by repository:\n'
+
+        for repo in language_repo:
+            if language_repo[repo]['total'] == 0:
+                continue
+
+            languages = {language:((100 * language_repo[repo][language])/language_repo[repo]['total']) for language in language_repo[repo]}
+            language_output += '\t' f'{repo}: {", ".join(["%s (%.2f%%)" % (k, languages[k]) for k in languages if k != "total"])}\n'
```
Comment (e-caste) [inline main.py]: Try using the limited float precision feature of fstrings
```suggestion
            language_output += f'\t{repo}: {", ".join(["{k} ({languages[k]:.2f}%)" for k in languages if k != "total"])}\n'
```

**Review by e-caste — CHANGES_REQUESTED** [review]
Some small changes required, nothing major. Working code is good, but readable working code is great (asd)

File: main.py (hunk 3 above)
Comment (alecello) [reply main.py]: Should be solved in d010048

File: main.py (hunk 2 above)
Comment (alecello) [reply main.py]: Should be resolved in 0ee80ba

**Review by e-caste — APPROVED** [review]

**Pull request merged**